### DISCUSSION TITLE
タメ口から敬語に変換するAPIをAzureのサービスを使って作りたい。

### DISCUSSION THREAD

**Discussion by ryutaro-kimura** [1ft-seabass/ms-azure-hackathon-league-2021#17] [Q&A] [answered]
## したいこと
タメ口から敬語に変換するAPIをAzureのサービスを使って作りたい。

## イメージ図(これはAzure使ってないやつ)
`curl` でAPIを叩いたらこんなresponsが来てほしい。
![image](https://user-images.githubusercontent.com/65991131/144708561-fbea0a94-3b7d-4cbd-a981-215b5bde3243.png)

API内部ではこうなってほしい。
![image](https://user-images.githubusercontent.com/65991131/144708468-b84bd0b0-eb71-452f-b327-46444735ef07.png)

## 知りたいこと
- Azureのどのサービスを使ったらできそうか
→以下記事を参考にLanguage APIで出来るかどうか調べたが、分からなかった。
https://docs.microsoft.com/ja-jp/azure/cognitive-services/what-are-cognitive-services

**Comment by 1ft-seabass** [comment, marked as answer]
- 1 音声を日本語の文章にする
- 2 その日本語の文章を分解する
  - おそらく形態素的なもの https://ja.wikipedia.org/wiki/%E5%BD%A2%E6%85%8B%E7%B4%A0
- 3 言葉の末尾を認識して内容は壊さず語尾を置き換える
  - 例 : したい → です。

おそらく、1 のところはできているので、2 と 3 で Azure を使いたいというイメージですね。

私の方もちょっと調べてみます。
